MaskedSelect.lua

```diff
@@ -0,0 +1,65 @@
+local MaskedSelect, parent = torch.class('nn.MaskedSelect', 'nn.Module')
+
+--[[ Sets the provided mask value for the module. ]]
+function MaskedSelect:__init()
+  parent.__init(self)
+  self._maskIndices = torch.LongTensor()
+  self._maskIndexBuffer = torch.LongTensor()
+  self._maskIndexBufferCPU = torch.FloatTensor()
+  self._gradBuffer = torch.Tensor()
+  self._gradMask = torch.ByteTensor()
+end
+
+--[[ Performs maskedSelect operation. ]]
+function MaskedSelect:updateOutput(input)
+  local input, mask = table.unpack(input)
```

Comment:
one last thing, I'm not sure that `table.unpack` exist in Lua 5.1. what people usually do is to add in the beginning of the file
```lua
local unpack = unpack or table.unpack
```
and then call `unpack` afterwards. Not sure we care that much about Lua 5.1 compatibility though (I'm thinking about https://github.com/torch/torch7/issues/516#issuecomment-176134155 )